[Bug]: workspaceCompilation not working with external dependencies

### AL-Go version

9.0

### Describe the issue

When I enable workspace compilation in AL-Go, the build fails with `AL1022` for external AL app dependencies.

The same repository builds successfully when `workspaceCompilation` is disabled, so this seems specific to the workspace compilation path rather than a general dependency or project configuration problem.

### Expected behavior

The build should succeed the same way it does without workspace compilation, or external dependencies should be resolved consistently between the normal build path and the workspace compilation path.

### Steps to reproduce

Enable workspace compilation in settings:

"workspaceCompilation": {
  "enabled": true,
  "parallelism": 4
}

### Additional context (logs, screenshots, etc.)

_No response_

@aholstrup1 maybe not the full raw logs but if you can tell which parts would help i can get them.

I did some checking myself and i can see from the logs that the dependency app is succesfully downloaded from   "appDependencyProbingPaths" and i can see that its loaded to C:\2\_work\repo_name and then later in the build its trying to get it from compiler/symbols:
"2026-04-20T06:18:28.4783905Z Compilation started for project 'AOR Test App1' containing '1' files at '06:18:10.546'.
2026-04-20T06:18:28.4788840Z 
2026-04-20T06:18:28.4821146Z ##[error]AL1022 A package with publisher 'Adaptor Oy', name 'Adaptor Foundation', and a version compatible with '1.7.0.0' could not be found in the package cache folders: C:\ProgramData\BcContainerHelper\compiler\bc24651370777compiler\symbols"

This is self-hosted runner and i checked the symbols folder and there is a zip file containing the .app file there.

Also tested this without "workspaceCompilation" and build completed.

@aholstrup1 
Yes if i disable workspace compilation, all works as expected

Under dependencies, it downloads a .zip file
Downloaded dependencies (Apps):
- C:\1\_work\AL-Go-Test\AL-Go-Test\.dependencies\AOR_Foundation-main-Apps-1.9.307.0.zip
Downloaded dependencies (Test Apps):
- (C:\1\_work\AL-Go-Test\AL-Go-Test\.dependencies\AOR_Foundation-main-TestApps-1.9.307.0.zip)

Here is the copied dependency part:
2026-04-20T09:49:04.3370542Z New-BcCompilerFolder Telemetry Correlation Id: 2f8f0d40-fc5a-45b9-8e59-1d4cf27effdb
2026-04-20T09:49:04.3449727Z ##[debug][AL-Go]Copied dependency app to package cache: AOR_Foundation-main-Apps-1.9.307.0.zip
2026-04-20T09:49:04.3845377Z ##[debug][AL-Go]Updating app.json at C:\1\_work\AL-Go-Test\AL-Go-Test\App1\app.json to version 1.0.5.2
2026-04-20T09:49:04.3904543Z ##[debug][AL-Go]Updated app.json at C:\1\_work\AL-Go-Test\AL-Go-Test\App1\app.json
2026-04-20T09:49:04.3930515Z ##[debug][AL-Go]Updating app.json at C:\1\_work\AL-Go-Test\AL-Go-Test\App2\app.json to version 1.0.5.2
2026-04-20T09:49:04.3959870Z ##[debug][AL-Go]Updated app.json at C:\1\_work\AL-Go-Test\AL-Go-Test\App2\app.json
2026-04-20T09:49:04.3986349Z ##[debug][AL-Go]Updating app.json at C:\1\_work\AL-Go-Test\AL-Go-Test\App3\app.json to version 1.0.5.2
2026-04-20T09:49:04.4015301Z ##[debug][AL-Go]Updated app.json at C:\1\_work\AL-Go-Test\AL-Go-Test\App3\app.json


Download Project Dependencies: logs:
Downloading project dependencies from probing paths
  Checking type
  Checking appFolders, testFolders and bcptTestFolders
  Application Dependency 24.0.0.0
  Updating app- and test Dependencies
  Analyzing Test App Dependencies
  No performance test apps found in bcptTestFolders in .AL-Go\settings.json
  Checking appDependencyProbingPaths
  Dependency to projects '*' in https://github.com/AdaptorOy/AOR_Foundation@main, version latest, release status latestBuild
  Using secret GHTOKENWORKFLOW for access to repository
  Locating Apps artifacts for projects: *
  ##[debug][AL-Go]Attempting to get Access Token but token is not in JSON format, returning as is.
  ##[debug][AL-Go]Attempting to get Access Token but token is not in JSON format, returning as is.
  ##[debug][AL-Go]Attempting to get Access Token but token is not in JSON format, returning as is.
  Finding latest successful CICD run for branch main in repository AdaptorOy/AOR_Foundation, checking last 90 days
  - https://api.github.com/repos/AdaptorOy/AOR_Foundation/actions/runs?per_page=100&page=1&exclude_pull_requests=true&status=completed&branch=main&created=>2026-01-20T09:47:00.9845529Z
  Last successful CICD run for branch main in repository AdaptorOy/AOR_Foundation is 24329886411 with SHA 7676fa0a840f374570fe05af53b2df18f035d944
  Getting artifacts for workflow run 24329886411, mask Apps, projects * and version *
  ##[debug][AL-Go]Attempting to get Access Token but token is not in JSON format, returning as is.
  Branch for workflow run 24329886411 is main
  Found artifact AOR_Foundation-main-Apps-1.9.307.0 (ID: 6400178652) for mask Apps and project *
  No more artifacts found for workflow run 24329886411
  Found 1 artifacts for mask Apps and projects * in workflow run 24329886411
  Downloading artifact AOR_Foundation-main-Apps-1.9.307.0
  https://api.github.com/repos/AdaptorOy/AOR_Foundation/actions/artifacts/6400178652/zip
  ##[debug][AL-Go]Attempting to get Access Token but token is not in JSON format, returning as is.
  Locating Dependencies artifacts for projects: *
  ##[debug][AL-Go]Attempting to get Access Token but token is not in JSON format, returning as is.
  ##[debug][AL-Go]Attempting to get Access Token but token is not in JSON format, returning as is.
  ##[debug][AL-Go]Attempting to get Access Token but token is not in JSON format, returning as is.
  Finding latest successful CICD run for branch main in repository AdaptorOy/AOR_Foundation, checking last 90 days
  - https://api.github.com/repos/AdaptorOy/AOR_Foundation/actions/runs?per_page=100&page=1&exclude_pull_requests=true&status=completed&branch=main&created=>2026-01-20T09:47:17.4632604Z
  Last successful CICD run for branch main in repository AdaptorOy/AOR_Foundation is 24329886411 with SHA 7676fa0a840f374570fe05af53b2df18f035d944
  Getting artifacts for workflow run 24329886411, mask Dependencies, projects * and version *
  ##[debug][AL-Go]Attempting to get Access Token but token is not in JSON format, returning as is.
  Branch for workflow run 24329886411 is main
  No more artifacts found for workflow run 24329886411
  Found 0 artifacts for mask Dependencies and projects * in workflow run 24329886411
  Could not find any Dependencies artifacts for projects *, version latest
  Locating TestApps artifacts for projects: *
  ##[debug][AL-Go]Attempting to get Access Token but token is not in JSON format, returning as is.
  ##[debug][AL-Go]Attempting to get Access Token but token is not in JSON format, returning as is.
  ##[debug][AL-Go]Attempting to get Access Token but token is not in JSON format, returning as is.
  Finding latest successful CICD run for branch main in repository AdaptorOy/AOR_Foundation, checking last 90 days
  - https://api.github.com/repos/AdaptorOy/AOR_Foundation/actions/runs?per_page=100&page=1&exclude_pull_requests=true&status=completed&branch=main&created=>2026-01-20T09:47:31.8791045Z
  Last successful CICD run for branch main in repository AdaptorOy/AOR_Foundation is 24329886411 with SHA 7676fa0a840f374570fe05af53b2df18f035d944
  Getting artifacts for workflow run 24329886411, mask TestApps, projects * and version *
  ##[debug][AL-Go]Attempting to get Access Token but token is not in JSON format, returning as is.
  Branch for workflow run 24329886411 is main
  Found artifact AOR_Foundation-main-TestApps-1.9.307.0 (ID: 6400179171) for mask TestApps and project *
  No more artifacts found for workflow run 24329886411
  Found 1 artifacts for mask TestApps and projects * in workflow run 24329886411
  Downloading artifact AOR_Foundation-main-TestApps-1.9.307.0
  https://api.github.com/repos/AdaptorOy/AOR_Foundation/actions/artifacts/6400179171/zip
  ##[debug][AL-Go]Attempting to get Access Token but token is not in JSON format, returning as is.
  ::endgroup::
::group::Downloading dependencies from settings (installApps and installTestApps)
Downloading dependencies from settings (installApps and installTestApps)
  No installApps or installTestApps settings found.
  ::endgroup::
Downloaded dependencies (Apps):
- C:\1\_work\AL-Go-Test\AL-Go-Test\.dependencies\AOR_Foundation-main-Apps-1.9.307.0.zip
Downloaded dependencies (Test Apps):
- (C:\1\_work\AL-Go-Test\AL-Go-Test\.dependencies\AOR_Foundation-main-TestApps-1.9.307.0.zip)
##[debug][AL-Go]Applying settings from ALGoOrgSettings (Variable)
##[debug][AL-Go]No settings found in .github\AL-Go-TemplateRepoSettings.doNotEdit.json (File)
##[debug][AL-Go]Applying settings from .github\AL-Go-Settings.json (File)
##[debug][AL-Go]No settings found in .github\AL-Go-TemplateProjectSettings.doNotEdit.json (File)
##[debug][AL-Go]Applying settings from .\.AL-Go\settings.json (File)
##[debug][AL-Go]No settings found in .github\CICD.settings.json (File)
##[debug][AL-Go]No settings found in .\.AL-Go\CICD.settings.json (File)
##[debug][AL-Go]No settings found in .\.AL-Go\youngdanger2.settings.json (File)
##[debug][AL-Go]Setting shell to powershell for non-ubuntu
##[debug][AL-Go]Setting gitHubRunnerShell to shell value: powershell
##[debug][AL-Go]Setting projectName to default value: .
Enabling Microsoft telemetry...
Enabling partner telemetry...
##[debug]Set output DownloadedApps = C:\1\_work\_temp\3gd1yv3h.4x3\DownloadedApps.json
##[debug]Set output DownloadedTestApps = C:\1\_work\_temp\3gd1yv3h.4x3\DownloadedTestApps.json
##[debug]Finished: run
##[debug]Evaluating: steps.DownloadDependencies.outputs.DownloadedApps
##[debug]Evaluating Index:
##[debug]..Evaluating Index:
##[debug]....Evaluating Index:
##[debug]......Evaluating steps:
##[debug]......=> Object
##[debug]......Evaluating String:
##[debug]......=> 'DownloadDependencies'
##[debug]....=> Object
##[debug]....Evaluating String:
##[debug]....=> 'outputs'
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'DownloadedApps'
##[debug]=> 'C:\1\_work\_temp\3gd1yv3h.4x3\DownloadedApps.json'
##[debug]Result: 'C:\1\_work\_temp\3gd1yv3h.4x3\DownloadedApps.json'
##[debug]Evaluating: steps.DownloadDependencies.outputs.DownloadedTestApps
##[debug]Evaluating Index:
##[debug]..Evaluating Index:
##[debug]....Evaluating Index:
##[debug]......Evaluating steps:
##[debug]......=> Object
##[debug]......Evaluating String:
##[debug]......=> 'DownloadDependencies'
##[debug]....=> Object
##[debug]....Evaluating String:
##[debug]....=> 'outputs'
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'DownloadedTestApps'
##[debug]=> 'C:\1\_work\_temp\3gd1yv3h.4x3\DownloadedTestApps.json'
##[debug]Result: 'C:\1\_work\_temp\3gd1yv3h.4x3\DownloadedTestApps.json'
##[debug]Finishing: Download Project Dependencies


Compile apps logs:
Checking appDependencyProbingPaths
Dependency to projects '*' in https://github.com/AdaptorOy/AOR_Foundation@main, version latest, release status latestBuild
Using secret GHTOKENWORKFLOW for access to repository
Download-Artifacts Telemetry Correlation Id: bc81541a-eb7e-4833-9e8f-7c97a6238ef6
using 7zip
Copying app files from C:\bcartifacts.cache\sandbox\28.0.46665.49267\fi\Extensions
PlatForm apps
- Microsoft_API Reports - Finance.app
- Microsoft__Exclude_APIV1_.app
- Microsoft__Exclude_APIV1_ Tests.app
- Microsoft__Exclude_APIV2_.app
- Microsoft__Exclude_APIV2_ Tests.app
- Microsoft_Application.app
- Microsoft_Application Test Library.app
- Microsoft_Audit File Export.app
- Microsoft_Audit File Export Tests.app
- Microsoft_Bank Account Reconciliation With AI.app
- Microsoft_Bank Account Reconciliation With AI Tests.app
- Microsoft__Exclude_Bank Deposits.app
- Microsoft__Exclude_Bank Deposits Tests.app
- Microsoft_Base Application.app
- Microsoft_Czech language (Czechia).app
- Microsoft_Danish language (Denmark).app
- Microsoft_Dutch language (Belgium).app
- Microsoft_Dutch language (Netherlands).app
- Microsoft_English language (Australia).app
- Microsoft_English language (Canada).app
- Microsoft_English language (New Zealand).app
- Microsoft_English language (United Kingdom).app
- Microsoft_English language (United States).app
- Microsoft_Finnish language (Finland).app
- Microsoft_French language (Belgium).app
- Microsoft_French language (Canada).app
- Microsoft_French language (France).app
- Microsoft_French language (Switzerland).app
- Microsoft_German language (Austria).app
- Microsoft_German language (Germany).app
- Microsoft_German language (Switzerland).app
- Microsoft_Icelandic language (Iceland).app
- Microsoft_Italian language (Italy).app
- Microsoft_Italian language (Switzerland).app
- Microsoft_Norwegian language (Norway).app
- Microsoft_Russian language (Russia).app
- Microsoft_Spanish language (Mexico).app
- Microsoft_Spanish language (Spain).app
- Microsoft_Swedish language (Sweden).app
- Microsoft_Library-NoTransactions.app
- Microsoft_Tests-Bank.app
- Microsoft_Tests-Cash Flow.app
- Microsoft_Tests-Cost Accounting.app
- Microsoft_Tests-CRM integration.app
- Microsoft_Tests-Data Exchange.app
- Microsoft_Tests-Dimension.app
- Microsoft_Tests-ERM.app
- Microsoft_Tests-Fixed Asset.app
- Microsoft_Tests-General Journal.app
- Microsoft_Tests-Graph.app
- Microsoft_Tests-Integration.app
- Microsoft_Tests-Job.app
- Microsoft_Tests-Local.app
- Microsoft_Tests-Marketing.app
- Microsoft_Tests-Misc.app
- Microsoft_Tests-Monitor Sensitive Fields.app
- Microsoft_Tests-Permissions.app
- Microsoft_Tests-Physical Inventory.app
- Microsoft_Tests-Prepayment.app
- Microsoft_Tests-Rapid Start.app
- Microsoft_Tests-Report.app
- Microsoft_Tests-Resource.app
- Microsoft_Tests-Reverse.app
- Microsoft_Tests-SCM-Assembly.app
- Microsoft_Tests-SCM-Manufacturing.app
- Microsoft_Tests-SCM-Service.app
- Microsoft_Tests-SCM.app
- Microsoft_Tests-SINGLESERVER.app
- Microsoft_Tests-SMB.app
- Microsoft_Tests-TestLibraries.app
- Microsoft_Tests-Upgrade.app
- Microsoft_Tests-User.app
- Microsoft_Tests-VAT.app
- Microsoft_Tests-Workflow.app
- Microsoft_Business Foundation.app
- Microsoft_Business Foundation Test Libraries.app
- Microsoft_Business Foundation Tests.app
- Microsoft__Exclude_ClientAddIns_.app
- Microsoft_Company Hub.app
- Microsoft_Contoso Coffee Demo Dataset.app
- Microsoft_Contoso Coffee Demo Dataset Tests.app
- Microsoft_Create Product Information With Copilot.app
- Microsoft_Data Archive.app
- Microsoft_Data Archive Tests.app
- Microsoft_Troubleshoot FA Ledger Entries.app
- Microsoft_Data Search.app
- Microsoft_Data Search Tests.app
- Microsoft_E-Document Core Demo Data.app
- Microsoft_E-Document Core.app
- Microsoft_E-Document Core Tests.app
- Microsoft_E-Document Connector - Avalara.app
- Microsoft_E-Document Connector - Avalara Tests.app
- Microsoft_E-Document Connector - B2Brouter.app
- Microsoft_E-Document Connector - B2Brouter Tests.app
- Microsoft_E-Document Connector - Continia.app
- Microsoft_E-Document Connector - Continia Tests.app
- Microsoft_E-Document Connector - ForNAV.app
- Microsoft_E-Document Connector - ForNAV Tests.app
- Microsoft_E-Document Connector - Logiq.app
- Microsoft_E-Document Connector - Logiq Tests.app
- Microsoft_E-Document Connector - Pagero.app
- Microsoft_E-Document Connector - Pagero Tests.app
- Microsoft_E-Document Connector - SignUp.app
- Microsoft_E-Document Connector - SignUp Tests.app
- Microsoft_Email - Current User Connector.app
- Microsoft_Email - Current User Connector Tests.app
- Microsoft_Email - Microsoft 365 Connector.app
- Microsoft_Email - Microsoft 365 Connector Tests.app
- Microsoft_Email - Outlook REST API.app
- Microsoft_Library Outlook REST API.app
- Microsoft_Email - SMTP API.app
- Microsoft_Email - SMTP API Tests.app
- Microsoft_Email - SMTP API Test Library.app
- Microsoft_Email - SMTP Connector.app
- Microsoft_Email - SMTP Connector Tests.app
- Microsoft__Exclude_Email Logging Using Graph API.app
- Microsoft__Exclude_Email Logging Using Graph API Tests.app
- Microsoft_Enforced Digital Vouchers.app
- Microsoft_Enforced Digital Vouchers Tests.app
- Microsoft_Enforced Digital Vouchers Test Library.app
- Microsoft_Error Messages with Recommendations.app
- Microsoft_Error Messages with Recommendations Tests.app
- Microsoft_ESG Statistical Accounts Demo Tool.app
- Microsoft_Essential Business Headlines.app
- Microsoft_Essential Business Headlines Test.app
- Microsoft_EU 3-Party Trade Purchase.app
- Microsoft_EU 3-Party Trade Purchase Tests.app
- Microsoft_Dynamics BC Excel Reports.app
- Microsoft_Dynamics BC Excel Reports Tests.app
- Microsoft_Excise Taxes.app
- Microsoft_Excise Taxes Tests.app
- Microsoft_External File Storage - Azure Blob Service Connector.app
- Microsoft_External File Storage - Azure File Service Connector.app
- Microsoft_External File Storage - SFTP Connector.app
- Microsoft_External File Storage - SharePoint Connector.app
- Microsoft_External Storage - Document Attachments.app
- Microsoft__Exclude_Business_Events_.app
- Microsoft__Exclude_Business_Events_Test.app
- Microsoft_Field Service Integration.app
- Microsoft_Field Service Integration Test.app
- Microsoft_Field Service Integration Test Library.app
- Microsoft_Intrastat Core Demo Data.app
- Microsoft_Intrastat Core.app
- Microsoft_Intrastat Core Tests.app
- Microsoft_Late Payment Prediction.app
- Microsoft_Late Payment Prediction Tests.app
- Microsoft_Manufacturing.app
- Microsoft_Manufacturing Tests.app
- Microsoft__Exclude_Master_Data_Management.app
- Microsoft__Exclude_Master_Data_Management_Tests.app
- Microsoft__Exclude_Master_Data_Management_Test_Library.app
- Microsoft_Universal Print Integration.app
- Microsoft_OnPrem Permissions.app
- Microsoft_OnPrem Permissions Tests.app
- Microsoft_Payment Practices.app
- Microsoft_Payment Practices Tests.app
- Microsoft_Payment Links to PayPal.app
- Microsoft_PayPal Payments Standard Tests.app
- Microsoft_PEPPOL.app
- Microsoft_PEPPOL Tests.app
- Microsoft__Exclude_PlanConfiguration_.app
- Microsoft__Exclude_PlanConfiguration_ Tests.app
- Microsoft_Power BI Report embeddings for Dynamics 365 Business Central.app
- Microsoft_PowerBI Reports Tests.app
- Microsoft_PowerBI Reports Test Library.app
- Microsoft_Quality Management.app
- Microsoft_Quality Management-Tests.app
- Microsoft_Quality Management Test Library.app
- Microsoft_Quality Management Contoso Coffee Demo Dataset.app
- Microsoft_Recommended Apps.app
- Microsoft_Recommended Apps Tests.app
- Microsoft__Exclude_ReportLayouts.app
- Microsoft_Report Layouts Tests.app
- Microsoft_Review General Ledger Entries.app
- Microsoft__Exclude_Review_General_Ledger_Entries_Tests.app
- Microsoft_SAF-T.app
- Microsoft_SAF-T Tests.app
- Microsoft_Sales and Inventory Forecast.app
- Microsoft_Sales and Inventory Forecast Tests.app
- Microsoft_Sales Lines Suggestions.app
- Microsoft_Sales Lines Suggestions Tests.app
- Microsoft_Send To Email Printer.app
- Microsoft_Service Declaration.app
- Microsoft_Service Declaration Tests.app
- Microsoft_Service Management.app
- Microsoft_Service Management Tests.app
- Microsoft_Shopify Connector.app
- Microsoft_Shopify Connector Test.app
- Microsoft_Simplified Bank Statement Import.app
- Microsoft_Simplified Bank Statement Import Test.app
- Microsoft_Statistical Accounts.app
- Microsoft_Statistical Accounts Tests.app
- Microsoft_Subscription Billing Demo Data.app
- Microsoft_Subscription Billing.app
- Microsoft_Subscription Billing Tests.app
- Microsoft_Sustainability.app
- Microsoft_Sustainability Tests.app
- Microsoft_Sustainability Contoso Coffee Demo Dataset.app
- Microsoft_Calculate Sustainability Emission with Copilot.app
- Microsoft_Sustainability Copilot Tests.app
- Microsoft_System Application.app
- Microsoft_System Application Partner Test.app
- Microsoft_System Application Test Library.app
- Microsoft_System Application Test.app
- Microsoft_AI Test Toolkit.app
- Microsoft_Performance Toolkit Samples.app
- Microsoft_Performance Toolkit Tests.app
- Microsoft_Performance Toolkit.app
- Microsoft_Any.app
- Microsoft_Library Assert.app
- Microsoft_Permissions Mock.app
- Microsoft_Library Variable Storage.app
- Microsoft_Test Runner.app
- Microsoft_Prevent Metadata Updates.app
- Microsoft_Send remittance advice by email.app
- Microsoft_VAT Group Management.app
- Microsoft_VAT Group Management Tests.app
- Microsoft_Withholding Tax.app
- Microsoft_Withholding Tax Tests.app
App apps
- Microsoft_AI Test Toolkit_28.0.46665.49267.app
- Microsoft_Any_28.0.46665.49267.app
- Microsoft_Application Test Library_28.0.46665.49267.app
- Microsoft_Application_28.0.46665.49267.app
- Microsoft_Base Application_28.0.46665.49267.app
- Microsoft_BuildTools_28.0.46665.49267.app
- Microsoft_Business Foundation Test Libraries_28.0.46665.49267.app
- Microsoft_Business Foundation_28.0.46665.49267.app
- Microsoft_DemoTool_28.0.46665.49267.app
- Microsoft_Library Assert_28.0.46665.49267.app
- Microsoft_Library Variable Storage_28.0.46665.49267.app
- Microsoft_Performance Toolkit Samples_28.0.46665.49267.app
- Microsoft_Performance Toolkit_28.0.46665.49267.app
- Microsoft_Permissions Mock_28.0.46665.49267.app
- Microsoft_System Application Test Library_28.0.46665.49267.app
- Microsoft_System Application_28.0.46665.49267.app
- Microsoft_Test Runner_28.0.46665.49267.app
- Microsoft_TestRunner-Internal_28.0.46665.49267.app
- Microsoft_Tests-Bank_28.0.46665.49267.app
- Microsoft_Tests-Cash Flow_28.0.46665.49267.app
- Microsoft_Tests-Cost Accounting_28.0.46665.49267.app
- Microsoft_Tests-CRM integration_28.0.46665.49267.app
- Microsoft_Tests-Data Exchange_28.0.46665.49267.app
- Microsoft_Tests-Dimension_28.0.46665.49267.app
- Microsoft_Tests-DotNet-Internal_28.0.46665.49267.app
- Microsoft_Tests-ERM_28.0.46665.49267.app
- Microsoft_Tests-Fixed Asset_28.0.46665.49267.app
- Microsoft_Tests-General Journal_28.0.46665.49267.app
- Microsoft_Tests-Graph_28.0.46665.49267.app
- Microsoft_Tests-Integration-Internal_28.0.46665.49267.app
- Microsoft_Tests-Integration_28.0.46665.49267.app
- Microsoft_Tests-Job_28.0.46665.49267.app
- Microsoft_Tests-Local_28.0.46665.49267.app
- Microsoft_Tests-Marketing_28.0.46665.49267.app
- Microsoft_Tests-Misc_28.0.46665.49267.app
- Microsoft_Tests-MOCKSERVICETESTS-Internal_28.0.46665.49267.app
- Microsoft_Tests-Monitor Sensitive Fields_28.0.46665.49267.app
- Microsoft_Tests-NewObjects-Internal_28.0.46665.49267.app
- Microsoft_Tests-Performance-Internal_28.0.46665.49267.app
- Microsoft_Tests-Permissions_28.0.46665.49267.app
- Microsoft_Tests-Physical Inventory_28.0.46665.49267.app
- Microsoft_Tests-Prepayment_28.0.46665.49267.app
- Microsoft_Tests-Rapid Start_28.0.46665.49267.app
- Microsoft_Tests-Report_28.0.46665.49267.app
- Microsoft_Tests-Resource_28.0.46665.49267.app
- Microsoft_Tests-Reverse_28.0.46665.49267.app
- Microsoft_Tests-SCM-Assembly_28.0.46665.49267.app
- Microsoft_Tests-SCM-Manufacturing_28.0.46665.49267.app
- Microsoft_Tests-SCM-Service_28.0.46665.49267.app
- Microsoft_Tests-SCM_28.0.46665.49267.app
- Microsoft_Tests-SINGLESERVER-Internal_28.0.46665.49267.app
- Microsoft_Tests-SINGLESERVER_28.0.46665.49267.app
- Microsoft_Tests-SMB_28.0.46665.49267.app
- Microsoft_Tests-TestLibraries_28.0.46665.49267.app
- Microsoft_Tests-Upgrade_28.0.46665.49267.app
- Microsoft_Tests-User_28.0.46665.49267.app
- Microsoft_Tests-VAT_28.0.46665.49267.app
- Microsoft_Tests-Workflow_28.0.46665.49267.app
Enumerating Apps in C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\symbols
Getting .app info C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\symbols\cache_AppInfo.json
Use C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\compiler\extension\bin\win32\altool.exe as altool executable (Exists = True).
- .\Microsoft_Agent Design Experience_28.0.46665.49267.app (using altool)
- .\Microsoft_Agent Samples_28.0.46665.49267.app (using altool)
- .\Microsoft_Agent Test Library_28.0.46665.49267.app (using altool)
- .\Microsoft_AI Development Toolkit_28.0.46665.49267.app (using altool)
- .\Microsoft_AI Test Toolkit_28.0.46665.49267.app (using altool)
- .\Microsoft_AMC Banking 365 Fundamentals_28.0.46665.49267.app (using altool)
- .\Microsoft_Any_28.0.46665.49267.app (using altool)
- .\Microsoft_API - Cross Environment Intercompany_28.0.46665.49267.app (using altool)
- .\Microsoft_API Reports - Finance_28.0.46665.49267.app (using altool)
- .\Microsoft_Application Test Library_28.0.46665.49267.app (using altool)
- .\Microsoft_Application_28.0.46665.49267.app (using altool)
- .\Microsoft_Audit File Export_28.0.46665.49267.app (using altool)
- .\Microsoft_Automatic Account Codes_28.0.46665.49267.app (using altool)
- .\Microsoft_Bank Account Reconciliation With AI_28.0.46665.49267.app (using altool)
- .\Microsoft_Base Application_28.0.46665.49267.app (using altool)
- .\Microsoft_Business Central Cloud Migration - Previous Release_28.0.46665.49267.app (using altool)
- .\Microsoft_Business Central Cloud Migration API_28.0.46665.49267.app (using altool)
- .\Microsoft_Business Central Intelligent Cloud_28.0.46665.49267.app (using altool)
- .\Microsoft_Business Foundation Test Libraries_28.0.46665.49267.app (using altool)
- .\Microsoft_Business Foundation_28.0.46665.49267.app (using altool)
- .\Microsoft_Calculate Sustainability Emission with Copilot_28.0.46665.49267.app (using altool)
- .\Microsoft_Company Hub_28.0.46665.49267.app (using altool)
- .\Microsoft_Contoso Coffee Demo Dataset (FI)_28.0.46665.49267.app (using altool)
- .\Microsoft_Contoso Coffee Demo Dataset_28.0.46665.49267.app (using altool)
- .\Microsoft_Create Product Information With Copilot_28.0.46665.49267.app (using altool)
- .\Microsoft_Czech language (Czechia)_28.0.46665.49267.app (using altool)
- .\Microsoft_Danish language (Denmark)_28.0.46665.49267.app (using altool)
- .\Microsoft_Data Archive_28.0.46665.49267.app (using altool)
- .\Microsoft_Data Search_28.0.46665.49267.app (using altool)
- .\Microsoft_Dutch language (Belgium)_28.0.46665.49267.app (using altool)
- .\Microsoft_Dutch language (Netherlands)_28.0.46665.49267.app (using altool)
- .\Microsoft_Dynamics BC Excel Reports_28.0.46665.49267.app (using altool)
- .\Microsoft_E-Document Connector - Avalara_28.0.46665.49267.app (using altool)
- .\Microsoft_E-Document Connector - B2Brouter_28.0.46665.49267.app (using altool)
- .\Microsoft_E-Document Connector - Continia_28.0.46665.49267.app (using altool)
- .\Microsoft_E-Document Connector - ForNAV_28.0.46665.49267.app (using altool)
- .\Microsoft_E-Document Connector - Logiq_28.0.46665.49267.app (using altool)
- .\Microsoft_E-Document Connector - Microsoft 365_28.0.46665.49267.app (using altool)
- .\Microsoft_E-Document Connector - Pagero_28.0.46665.49267.app (using altool)
- .\Microsoft_E-Document Connector - SignUp_28.0.46665.49267.app (using altool)
- .\Microsoft_E-Document Core Demo Data FI_28.0.46665.49267.app (using altool)
- .\Microsoft_E-Document Core Demo Data_28.0.46665.49267.app (using altool)
- .\Microsoft_E-Document Core_28.0.46665.49267.app (using altool)
- .\Microsoft_Email - Current User Connector_28.0.46665.49267.app (using altool)
- .\Microsoft_Email - Microsoft 365 Connector_28.0.46665.49267.app (using altool)
- .\Microsoft_Email - Outlook REST API_28.0.46665.49267.app (using altool)
- .\Microsoft_Email - SMTP API_28.0.46665.49267.app (using altool)
- .\Microsoft_Email - SMTP Connector_28.0.46665.49267.app (using altool)
- .\Microsoft_Enforced Digital Vouchers_28.0.46665.49267.app (using altool)
- .\Microsoft_English language (Australia)_28.0.46665.49267.app (using altool)
- .\Microsoft_English language (Canada)_28.0.46665.49267.app (using altool)
- .\Microsoft_English language (New Zealand)_28.0.46665.49267.app (using altool)
- .\Microsoft_English language (United Kingdom)_28.0.46665.49267.app (using altool)
- .\Microsoft_English language (United States)_28.0.46665.49267.app (using altool)
- .\Microsoft_Error Messages with Recommendations_28.0.46665.49267.app (using altool)
- .\Microsoft_ESG Statistical Accounts Demo Tool_28.0.46665.49267.app (using altool)
- .\Microsoft_Essential Business Headlines_28.0.46665.49267.app (using altool)
- .\Microsoft_EU 3-Party Trade Purchase_28.0.46665.49267.app (using altool)
- .\Microsoft_Excise Taxes_28.0.46665.49267.app (using altool)
- .\Microsoft_External File Storage - Azure Blob Service Connector_28.0.46665.49267.app (using altool)
- .\Microsoft_External File Storage - Azure File Service Connector_28.0.46665.49267.app (using altool)
- .\Microsoft_External File Storage - SFTP Connector_28.0.46665.49267.app (using altool)
- .\Microsoft_External File Storage - SharePoint Connector_28.0.46665.49267.app (using altool)
- .\Microsoft_External Storage - Document Attachments_28.0.46665.49267.app (using altool)
- .\Microsoft_FI Core_28.0.46665.49267.app (using altool)
- .\Microsoft_Field Service Integration_28.0.46665.49267.app (using altool)
- .\Microsoft_Finnish language (Finland)_28.0.46665.49267.app (using altool)
- .\Microsoft_French language (Belgium)_28.0.46665.49267.app (using altool)
- .\Microsoft_French language (Canada)_28.0.46665.49267.app (using altool)
- .\Microsoft_French language (France)_28.0.46665.49267.app (using altool)
- .\Microsoft_French language (Switzerland)_28.0.46665.49267.app (using altool)
- .\Microsoft_German language (Austria)_28.0.46665.49267.app (using altool)
- .\Microsoft_German language (Germany)_28.0.46665.49267.app (using altool)
- .\Microsoft_German language (Switzerland)_28.0.46665.49267.app (using altool)
- .\Microsoft_Icelandic language (Iceland)_28.0.46665.49267.app (using altool)
- .\Microsoft_Intelligent Cloud Base_28.0.46665.49267.app (using altool)
- .\Microsoft_Intrastat Core Demo Data_28.0.46665.49267.app (using altool)
- .\Microsoft_Intrastat Core_28.0.46665.49267.app (using altool)
- .\Microsoft_Intrastat FI_28.0.46665.49267.app (using altool)
- .\Microsoft_Italian language (Italy)_28.0.46665.49267.app (using altool)
- .\Microsoft_Italian language (Switzerland)_28.0.46665.49267.app (using altool)
- .\Microsoft_Late Payment Prediction_28.0.46665.49267.app (using altool)
- .\Microsoft_Library Assert_28.0.46665.49267.app (using altool)
- .\Microsoft_Library Variable Storage_28.0.46665.49267.app (using altool)
- .\Microsoft_Manufacturing_28.0.46665.49267.app (using altool)
- .\Microsoft_Microsoft Pay Payments_28.0.46665.49267.app (using altool)
- .\Microsoft_Norwegian language (Norway)_28.0.46665.49267.app (using altool)
- .\Microsoft_Payables Agent_28.0.46665.49267.app (using altool)
- .\Microsoft_Payment Links to PayPal_28.0.46665.49267.app (using altool)
- .\Microsoft_Payment Practices_28.0.46665.49267.app (using altool)
- .\Microsoft_PEPPOL_28.0.46665.49267.app (using altool)
- .\Microsoft_Performance Toolkit Samples_28.0.46665.49267.app (using altool)
- .\Microsoft_Performance Toolkit Tests.app (using altool)
- .\Microsoft_Performance Toolkit_28.0.46665.49267.app (using altool)
- .\Microsoft_Permissions Mock_28.0.46665.49267.app (using altool)
- .\Microsoft_Power BI Report embeddings for Dynamics 365 Business Central_28.0.46665.49267.app (using altool)
- .\Microsoft_Quality Management Contoso Coffee Demo Dataset_28.0.46665.49267.app (using altool)
- .\Microsoft_Quality Management_28.0.46665.49267.app (using altool)
- .\Microsoft_Recommended Apps_28.0.46665.49267.app (using altool)
- .\Microsoft_Review General Ledger Entries_28.0.46665.49267.app (using altool)
- .\Microsoft_Russian language (Russia)_28.0.46665.49267.app (using altool)
- .\Microsoft_Sales and Inventory Forecast_28.0.46665.49267.app (using altool)
- .\Microsoft_Sales Lines Suggestions_28.0.46665.49267.app (using altool)
- .\Microsoft_Sales Order Agent_28.0.46665.49267.app (using altool)
- .\Microsoft_Send To Email Printer_28.0.46665.49267.app (using altool)
- .\Microsoft_Service Management_28.0.46665.49267.app (using altool)
- .\Microsoft_Shopify Connector_28.0.46665.49267.app (using altool)
- .\Microsoft_Simplified Bank Statement Import_28.0.46665.49267.app (using altool)
- .\Microsoft_Spanish language (Mexico)_28.0.46665.49267.app (using altool)
- .\Microsoft_Spanish language (Spain)_28.0.46665.49267.app (using altool)
- .\Microsoft_Statistical Accounts_28.0.46665.49267.app (using altool)
- .\Microsoft_Subscription Billing Demo Data_28.0.46665.49267.app (using altool)
- .\Microsoft_Subscription Billing_28.0.46665.49267.app (using altool)
- .\Microsoft_Sustainability Contoso Coffee Demo Dataset_28.0.46665.49267.app (using altool)
- .\Microsoft_Sustainability_28.0.46665.49267.app (using altool)
- .\Microsoft_Swedish language (Sweden)_28.0.46665.49267.app (using altool)
- .\Microsoft_System Application Test Library_28.0.46665.49267.app (using altool)
- .\Microsoft_System Application_28.0.46665.49267.app (using altool)
- .\Microsoft_Test Runner_28.0.46665.49267.app (using altool)
- .\Microsoft_TestRunner-Internal_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Bank_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Cash Flow_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Cost Accounting_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-CRM integration_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Data Exchange_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Dimension_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-DotNet-Internal_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-ERM_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Fixed Asset_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-General Journal_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Graph_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Integration-Internal_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Integration_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Job_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Local_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Marketing_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Misc_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-MOCKSERVICETESTS-Internal_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Monitor Sensitive Fields_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-NewObjects-Internal_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Performance-Internal_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Permissions_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Physical Inventory_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Prepayment_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Rapid Start_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Report_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Resource_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Reverse_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-SCM-Assembly_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-SCM-Manufacturing_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-SCM-Service_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-SCM_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-SINGLESERVER-Internal_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-SINGLESERVER_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-SMB_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-TestLibraries_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Upgrade_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-User_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-VAT_28.0.46665.49267.app (using altool)
- .\Microsoft_Tests-Workflow_28.0.46665.49267.app (using altool)
- .\Microsoft_Troubleshoot FA Ledger Entries_28.0.46665.49267.app (using altool)
- .\Microsoft_Universal Print Integration_28.0.46665.49267.app (using altool)
- .\Microsoft_VAT Group Management_28.0.46665.49267.app (using altool)
- .\Microsoft_Withholding Tax_28.0.46665.49267.app (using altool)
- .\Microsoft__Exclude_APIV1__28.0.46665.49267.app (using altool)
- .\Microsoft__Exclude_APIV2__28.0.46665.49267.app (using altool)
- .\Microsoft__Exclude_Bank Deposits_28.0.46665.49267.app (using altool)
- .\Microsoft__Exclude_Business_Events__28.0.46665.49267.app (using altool)
- .\Microsoft__Exclude_ClientAddIns__28.0.46665.49267.app (using altool)
- .\Microsoft__Exclude_Connectivity Apps_28.0.46665.49267.app (using altool)
- .\Microsoft__Exclude_Email Logging Using Graph API_28.0.46665.49267.app (using altool)
- .\Microsoft__Exclude_Master_Data_Management_28.0.46665.49267.app (using altool)
- .\Microsoft__Exclude_Microsoft Dynamics 365 - SmartList_28.0.46665.49267.app (using altool)
- .\Microsoft__Exclude_Onboarding Signals_28.0.46665.49267.app (using altool)
- .\Microsoft__Exclude_PlanConfiguration__28.0.46665.49267.app (using altool)
- .\Microsoft__Exclude_ReportLayouts_28.0.46665.49267.app (using altool)
- .\Microsoft__Exclude_SyncBaseApp__28.0.46665.49267.app (using altool)
- .\System.app (using altool)
New-BcCompilerFolder Telemetry Correlation Id: 2f8f0d40-fc5a-45b9-8e59-1d4cf27effdb
##[debug][AL-Go]Copied dependency app to package cache: AOR_Foundation-main-Apps-1.9.307.0.zip
##[debug][AL-Go]Updating app.json at C:\1\_work\AL-Go-Test\AL-Go-Test\App1\app.json to version 1.0.5.2
##[debug][AL-Go]Updated app.json at C:\1\_work\AL-Go-Test\AL-Go-Test\App1\app.json
##[debug][AL-Go]Updating app.json at C:\1\_work\AL-Go-Test\AL-Go-Test\App2\app.json to version 1.0.5.2
##[debug][AL-Go]Updated app.json at C:\1\_work\AL-Go-Test\AL-Go-Test\App2\app.json
##[debug][AL-Go]Updating app.json at C:\1\_work\AL-Go-Test\AL-Go-Test\App3\app.json to version 1.0.5.2
##[debug][AL-Go]Updated app.json at C:\1\_work\AL-Go-Test\AL-Go-Test\App3\app.json
##[debug][AL-Go]Determining assembly probing paths...
Probing Paths:
- C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\OpenXML
- C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.39
- C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\6.0.39
- C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\Service
- C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\Mock Assemblies
Executing: C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\compiler\extension\bin\win32\altool.exe workspace create C:\1\_work\AL-Go-Test\AL-Go-Test\tempWorkspace20260420094904.code-workspace .\App1 .\App2 .\App3
##[debug][AL-Go]Workspace created at C:\1\_work\AL-Go-Test\AL-Go-Test\tempWorkspace20260420094904.code-workspace
Executing: C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\compiler\extension\bin\win32\altool.exe workspace compile C:\1\_work\AL-Go-Test\AL-Go-Test\tempWorkspace20260420094904.code-workspace --maxcpucount 4 --packagecachepath C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\symbols --outfolder C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\symbols --assemblyprobingpaths C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\OpenXML C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.39 C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\6.0.39 C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\Service C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\Mock Assemblies --analyzers CodeCop,PTECop --sourcerepositoryurl https://github.com/AdaptorOy/AL-Go-Test --sourcecommit f590ae0ed868d38a8137f9ff5655a7d66ec34b14 --loglevel Warning --logdirectory C:\1\_work\AL-Go-Test\AL-Go-Test\.\.buildartifacts


Unable to compile 2 remaining project(s) due to failed dependencies or circular references:
  - AOR Test App3 (77d6b107-d9c2-4cb1-b838-dd744c9dd320)
  - AOR Test App2 (cc1641c4-b91d-4a38-83a9-23ea1e58945d)


================================================================================
COMPILATION SUMMARY
================================================================================
Total Duration: 00:00:17
Total Projects: 3
Succeeded:      0
Failed:         1
--------------------------------------------------------------------------------

Failed Projects:
  ✗ AOR Test App1 (17.5s)
    Log: C:\1\_work\AL-Go-Test\AL-Go-Test\.\.buildartifacts\AOR Test App1_20260420_094905.log
================================================================================
Error during compilation: C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\compiler\extension\bin\win32\altool.exe workspace compile C:\1\_work\AL-Go-Test\AL-Go-Test\tempWorkspace20260420094904.code-workspace --maxcpucount 4 --packagecachepath C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\symbols --outfolder C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\symbols --assemblyprobingpaths C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\OpenXML C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.39 C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\6.0.39 C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\Service C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\Mock Assemblies --analyzers CodeCop,PTECop --sourcerepositoryurl https://github.com/AdaptorOy/AL-Go-Test --sourcecommit f590ae0ed868d38a8137f9ff5655a7d66ec34b14 --loglevel Warning --logdirectory C:\1\_work\AL-Go-Test\AL-Go-Test\.\.buildartifacts failed with exit code 1
::group::Build Logs
Build Logs
  Compilation started at: 2026-04-20 09:49:05
  Project: AOR Test App1
  Path: C:\1\_work\AL-Go-Test\AL-Go-Test\App1
  Command: C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\compiler\extension\bin\win32\alc.exe  -packagecachepath:"C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\symbols" -outfolder:"C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\symbols" -assemblyprobingpaths:"C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\OpenXML","C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.39","C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\6.0.39","C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\Service","C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\Mock Assemblies" -a:"C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\compiler\extension\bin\win32\../Analyzers\Microsoft.Dynamics.Nav.Analyzers.Common.dll" -a:"C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\compiler\extension\bin\win32\../Analyzers\Microsoft.Dynamics.Nav.CodeCop.dll" -a:"C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\compiler\extension\bin\win32\../Analyzers\Microsoft.Dynamics.Nav.PerTenantExtensionCop.dll" -generatereportlayout- -sourcerepositoryurl:"https://github.com/AdaptorOy/AL-Go-Test" -sourcecommit:"f590ae0ed868d38a8137f9ff5655a7d66ec34b14" -project:"C:\1\_work\AL-Go-Test\AL-Go-Test\App1"
  ================================================================================
  
  Microsoft (R) AL Compiler version 17.0.35.21536
  Copyright (C) Microsoft Corporation. All rights reserved
  
  Compilation started for project 'AOR Test App1' containing '1' files at '09:49:05.405'.
  
  Error: AL1022 A package with publisher 'Adaptor Oy', name 'Adaptor Foundation', and a version compatible with '1.7.0.0' could not be found in the package cache folders: C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\symbols
  
  Compilation ended at '09:49:22.615'.
  
  
  ================================================================================
  Compilation ended at: 2026-04-20 09:49:22
  Exit code: 1
  Convert-AlcOutputToAzureDevOps Telemetry Correlation Id: 14b397e7-fd06-438f-8f53-3b0c7c406c20
  ::endgroup::
C:\1\_work\AL-Go-Test\AL-Go-Test\.\BuildOutput.txt
##[debug][AL-Go]Applying settings from ALGoOrgSettings (Variable)
##[debug][AL-Go]No settings found in .github\AL-Go-TemplateRepoSettings.doNotEdit.json (File)
##[debug][AL-Go]Applying settings from .github\AL-Go-Settings.json (File)
##[debug][AL-Go]No settings found in .github\AL-Go-TemplateProjectSettings.doNotEdit.json (File)
##[debug][AL-Go]Applying settings from .\.AL-Go\settings.json (File)
##[debug][AL-Go]No settings found in .github\CICD.settings.json (File)
##[debug][AL-Go]No settings found in .\.AL-Go\CICD.settings.json (File)
##[debug][AL-Go]No settings found in .\.AL-Go\youngdanger2.settings.json (File)
##[debug][AL-Go]Setting shell to powershell for non-ubuntu
##[debug][AL-Go]Setting gitHubRunnerShell to shell value: powershell
##[debug][AL-Go]Setting projectName to default value: .
Enabling Microsoft telemetry...
Enabling partner telemetry...
Error: Unexpected error when running action. Error Message: C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\compiler\extension\bin\win32\altool.exe workspace compile C:\1\_work\AL-Go-Test\AL-Go-Test\tempWorkspace20260420094904.code-workspace --maxcpucount 4 --packagecachepath C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\symbols --outfolder C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\symbols --assemblyprobingpaths C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\OpenXML C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.39 C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\6.0.39 C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\Service C:\ProgramData\BcContainerHelper\compiler\bc24654471906compiler\dlls\Mock Assemblies --analyzers CodeCop,PTECop --sourcerepositoryurl https://github.com/AdaptorOy/AL-Go-Test --sourcecommit f590ae0ed868d38a8137f9ff5655a7d66ec34b14 --loglevel Warning --logdirectory C:\1\_work\AL-Go-Test\AL-Go-Test\.\.buildartifacts failed with exit code 1, StackTrace: at RunAndCheck, C:\1\_work\_actions\microsoft\AL-Go-Actions\v9.0\AL-Go-Helper.ps1: line 2435 <- at CompileAppsInWorkspace, C:\1\_work\_actions\microsoft\AL-Go-Actions\v9.0\.Modules\CompileFromWorkspace.psm1: line 550 <- at Build-AppsInWorkspace, C:\1\_work\_actions\microsoft\AL-Go-Actions\v9.0\.Modules\CompileFromWorkspace.psm1: line 321 <- at <ScriptBlock>, C:\1\_work\_actions\microsoft\AL-Go-Actions\v9.0\CompileApps\Compile.ps1: line 160 <- at <ScriptBlock>, C:\1\_work\_temp\43f742de-f3a1-43c6-bb8e-d4d84de94769.ps1: line 3 <- at <ScriptBlock>, C:\1\_work\_actions\microsoft\AL-Go-Actions\v9.0\Invoke-AlGoAction.ps1: line 21 <- at <ScriptBlock>, C:\1\_work\_temp\43f742de-f3a1-43c6-bb8e-d4d84de94769.ps1: line 2 <- at <ScriptBlock>, <No file>: line 1
Error: Process completed with exit code 1.
##[debug]Finished: run
##[debug]Finishing: Compile Apps

--- a/Actions/CompileApps/Compile.ps1
+++ b/Actions/CompileApps/Compile.ps1
@@ -104,12 +104,15 @@ try {
     $compilerFolder = New-BcCompilerFolder -artifactUrl $artifact -vsixFile $settings.vsixFile -containerName "$($containerName)compiler" -cacheFolder $cacheFolder
     $packageCachePath = Join-Path $compilerFolder "symbols"
 
-    # Copy dependency apps to the package cache so the compiler can resolve them
-    foreach ($appFile in $dependencyApps) {
+    # Copy dependency apps and test apps to the package cache so the compiler can resolve them
+    foreach ($appFile in ($dependencyApps + $dependencyTestApps)) {
         $appFile = $appFile.Trim('()')
         if ($appFile -and (Test-Path $appFile)) {
             Copy-Item -Path $appFile -Destination $packageCachePath -Force
             OutputDebug "Copied dependency app to package cache: $(Split-Path $appFile -Leaf)"
+        }
+        elseif ($appFile) {
+            OutputWarning -message "Dependency app file not found: $appFile"
         }
     }
 

--- a/Actions/DownloadProjectDependencies/DownloadProjectDependencies.Action.ps1
+++ b/Actions/DownloadProjectDependencies/DownloadProjectDependencies.Action.ps1
@@ -27,17 +27,7 @@ function DownloadDependenciesFromProbingPaths {
 
         # GetDependencies may return .zip files (from DownloadArtifact/DownloadRelease).
         # Extract .app files from any zips so downstream consumers receive clean .app paths.
-        return @($dependencies | ForEach-Object {
-            $isTestApp = $_.StartsWith('(')
-            $filePath = $_.Trim('()')
-            if ($filePath -and (Test-Path $filePath) -and (Test-IsZipFile -Path $filePath)) {
-                $appFiles = Expand-ZipFileToAppFiles -ZipFile $filePath -DestinationPath $destinationPath
-                if ($isTestApp) { $appFiles | ForEach-Object { "($_)" } } else { $appFiles }
-            }
-            else {
-                $_
-            }
-        })
+        return Resolve-DependencyFiles -Dependencies $dependencies -DestinationPath $destinationPath
     }
 }
 

--- a/Actions/DownloadProjectDependencies/DownloadProjectDependencies.Action.ps1
+++ b/Actions/DownloadProjectDependencies/DownloadProjectDependencies.Action.ps1
@@ -111,7 +111,7 @@ function DownloadDependenciesFromCurrentBuild {
         }
     }
 
-    return $downloadedDependencies
+    return Resolve-DependencyFiles -Dependencies $downloadedDependencies -DestinationPath $destinationPath
 }
 
 . (Join-Path -Path $PSScriptRoot -ChildPath "..\AL-Go-Helper.ps1" -Resolve)

--- a/Actions/DownloadProjectDependencies/DownloadProjectDependencies.Action.ps1
+++ b/Actions/DownloadProjectDependencies/DownloadProjectDependencies.Action.ps1
@@ -23,7 +23,21 @@ function DownloadDependenciesFromProbingPaths {
     $settings = AnalyzeRepo -settings $settings -baseFolder $baseFolder -project $project -doNotCheckArtifactSetting -doNotIssueWarnings
     $settings = CheckAppDependencyProbingPaths -settings $settings -token $token -baseFolder $baseFolder -project $project
     if ($settings.ContainsKey('appDependencyProbingPaths') -and $settings.appDependencyProbingPaths) {
-        return GetDependencies -probingPathsJson $settings.appDependencyProbingPaths -saveToPath $destinationPath | Where-Object { $_ }
+        $dependencies = GetDependencies -probingPathsJson $settings.appDependencyProbingPaths -saveToPath $destinationPath | Where-Object { $_ }
+
+        # GetDependencies may return .zip files (from DownloadArtifact/DownloadRelease).
+        # Extract .app files from any zips so downstream consumers receive clean .app paths.
+        return @($dependencies | ForEach-Object {
+            $isTestApp = $_.StartsWith('(')
+            $filePath = $_.Trim('()')
+            if ($filePath -and (Test-Path $filePath) -and (Test-IsZipFile -Path $filePath)) {
+                $appFiles = Expand-ZipFileToAppFiles -ZipFile $filePath -DestinationPath $destinationPath
+                if ($isTestApp) { $appFiles | ForEach-Object { "($_)" } } else { $appFiles }
+            }
+            else {
+                $_
+            }
+        })
     }
 }
 
@@ -148,7 +162,7 @@ $downloadedDependencies | ForEach-Object {
     }
 }
 
-# Add dependencies from settings
+# Add dependencies from settings (these are already resolved to .app files)
 $downloadedApps += $settingsDependencies.Apps
 $downloadedTestApps += $settingsDependencies.TestApps
 

--- a/Actions/DownloadProjectDependencies/DownloadProjectDependencies.psm1
+++ b/Actions/DownloadProjectDependencies/DownloadProjectDependencies.psm1
@@ -328,4 +328,45 @@ function Get-DependenciesFromInstallApps {
     return $install
 }
 
-Export-ModuleMember -Function Get-AppFilesFromUrl, Get-AppFilesFromLocalPath, Get-DependenciesFromInstallApps, Expand-ZipFileToAppFiles, Test-IsZipFile
+<#
+    .SYNOPSIS
+    Resolves dependency file paths by extracting .app files from any zip archives.
+    .DESCRIPTION
+    Takes an array of dependency file paths (which may be .app files or .zip archives)
+    and returns an array of .app file paths. Zip archives are extracted and the contained
+    .app files are copied to the destination path. Test app markers (parentheses wrapping)
+    are preserved through extraction.
+    .PARAMETER Dependencies
+    An array of dependency file paths. Test apps are wrapped in parentheses, e.g. "(path.zip)".
+    .PARAMETER DestinationPath
+    The path where extracted .app files should be placed.
+    .OUTPUTS
+    An array of resolved .app file paths, with test app markers preserved.
+#>
+function Resolve-DependencyFiles {
+    Param(
+        [Parameter(Mandatory = $false)]
+        [string[]] $Dependencies = @(),
+        [Parameter(Mandatory = $true)]
+        [string] $DestinationPath
+    )
+
+    if (-not $Dependencies -or $Dependencies.Count -eq 0) {
+        return @()
+    }
+
+    return @($Dependencies | ForEach-Object {
+        $isTestApp = $_.StartsWith('(')
+        $filePath = $_.Trim('()')
+        if ($filePath -and (Test-Path $filePath) -and (Test-IsZipFile -Path $filePath)) {
+            $appFiles = Expand-ZipFileToAppFiles -ZipFile $filePath -DestinationPath $DestinationPath
+            Remove-Item -Path $filePath -Force -ErrorAction SilentlyContinue
+            if ($isTestApp) { $appFiles | ForEach-Object { "($_)" } } else { $appFiles }
+        }
+        else {
+            $_
+        }
+    })
+}
+
+Export-ModuleMember -Function Get-AppFilesFromUrl, Get-AppFilesFromLocalPath, Get-DependenciesFromInstallApps, Resolve-DependencyFiles

--- a/Actions/DownloadProjectDependencies/DownloadProjectDependencies.psm1
+++ b/Actions/DownloadProjectDependencies/DownloadProjectDependencies.psm1
@@ -328,4 +328,4 @@ function Get-DependenciesFromInstallApps {
     return $install
 }
 
-Export-ModuleMember -Function Get-AppFilesFromUrl, Get-AppFilesFromLocalPath, Get-DependenciesFromInstallApps
+Export-ModuleMember -Function Get-AppFilesFromUrl, Get-AppFilesFromLocalPath, Get-DependenciesFromInstallApps, Expand-ZipFileToAppFiles, Test-IsZipFile

--- a/RELEASENOTES.md
+++ b/RELEASENOTES.md
@@ -1,6 +1,7 @@
 ### Issues
 
 - Issue 2204 - Workspace compilation ignores vsixFile setting
+- Issue 2214 - Workspace compilation not working with external dependencies
 
 ## v9.0
 

--- a/Tests/DownloadProjectDependencies.Test.ps1
+++ b/Tests/DownloadProjectDependencies.Test.ps1
@@ -530,3 +530,109 @@ Describe "DownloadProjectDependencies - Get-DependenciesFromInstallApps Tests" {
         { Get-DependenciesFromInstallApps -DestinationPath $downloadPath } | Should -Throw "*unknown secret 'missingSecret'*"
     }
 }
+
+Describe "DownloadProjectDependencies - Resolve-DependencyFiles Tests" {
+    BeforeEach {
+        [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', 'testFolder', Justification = 'False positive.')]
+        $testFolder = (New-Item -ItemType Directory -Path (Join-Path $([System.IO.Path]::GetTempPath()) $([System.IO.Path]::GetRandomFileName()))).FullName
+        [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', 'destFolder', Justification = 'False positive.')]
+        $destFolder = (New-Item -ItemType Directory -Path (Join-Path $testFolder "dest")).FullName
+    }
+
+    AfterEach {
+        if (Test-Path $testFolder) {
+            Remove-Item -Path $testFolder -Recurse -Force
+        }
+    }
+
+    It 'Returns empty array for empty input' {
+        $result = Resolve-DependencyFiles -Dependencies @() -DestinationPath $destFolder
+        $result | Should -HaveCount 0
+    }
+
+    It 'Returns empty array for null input' {
+        $result = Resolve-DependencyFiles -Dependencies $null -DestinationPath $destFolder
+        $result | Should -HaveCount 0
+    }
+
+    It 'Passes through .app file paths unchanged' {
+        $appFile = Join-Path $testFolder "myapp.app"
+        Set-Content -Path $appFile -Value "fake app content"
+
+        $result = @(Resolve-DependencyFiles -Dependencies @($appFile) -DestinationPath $destFolder)
+
+        $result | Should -HaveCount 1
+        $result[0] | Should -Be $appFile
+    }
+
+    It 'Passes through test app markers for .app files' {
+        $appFile = Join-Path $testFolder "testapp.app"
+        Set-Content -Path $appFile -Value "fake test app"
+
+        $result = @(Resolve-DependencyFiles -Dependencies @("($appFile)") -DestinationPath $destFolder)
+
+        $result | Should -HaveCount 1
+        $result[0] | Should -Be "($appFile)"
+    }
+
+    It 'Extracts .app files from zip archives' {
+        $appFile = Join-Path $testFolder "Foundation_1.0.0.0.app"
+        Set-Content -Path $appFile -Value "fake app content"
+        $zipFile = Join-Path $testFolder "Foundation-main-Apps-1.0.0.0.zip"
+        Compress-Archive -Path $appFile -DestinationPath $zipFile
+
+        $result = @(Resolve-DependencyFiles -Dependencies @($zipFile) -DestinationPath $destFolder)
+
+        $result | Should -HaveCount 1
+        $result[0] | Should -BeLike "*Foundation_1.0.0.0.app"
+        Test-Path $result[0] | Should -Be $true
+    }
+
+    It 'Removes source zip after extraction' {
+        $appFile = Join-Path $testFolder "app.app"
+        Set-Content -Path $appFile -Value "fake"
+        $zipFile = Join-Path $testFolder "deps.zip"
+        Compress-Archive -Path $appFile -DestinationPath $zipFile
+
+        Resolve-DependencyFiles -Dependencies @($zipFile) -DestinationPath $destFolder
+
+        Test-Path $zipFile | Should -Be $false
+    }
+
+    It 'Preserves test app markers when extracting zips' {
+        $appFile = Join-Path $testFolder "testlib.app"
+        Set-Content -Path $appFile -Value "fake test lib"
+        $zipFile = Join-Path $testFolder "TestApps-1.0.0.0.zip"
+        Compress-Archive -Path $appFile -DestinationPath $zipFile
+
+        $result = @(Resolve-DependencyFiles -Dependencies @("($zipFile)") -DestinationPath $destFolder)
+
+        $result | Should -HaveCount 1
+        $result[0] | Should -Match '^\('
+        $result[0] | Should -Match '\)$'
+        $result[0].Trim('()') | Should -BeLike "*testlib.app"
+    }
+
+    It 'Handles mixed .app and .zip dependencies' {
+        $appFile = Join-Path $testFolder "direct.app"
+        Set-Content -Path $appFile -Value "direct app"
+
+        $zippedApp = Join-Path $testFolder "zipped.app"
+        Set-Content -Path $zippedApp -Value "zipped app"
+        $zipFile = Join-Path $testFolder "deps.zip"
+        Compress-Archive -Path $zippedApp -DestinationPath $zipFile
+
+        $result = @(Resolve-DependencyFiles -Dependencies @($appFile, $zipFile) -DestinationPath $destFolder)
+
+        $result | Should -HaveCount 2
+        $result[0] | Should -Be $appFile
+        $result[1] | Should -BeLike "*zipped.app"
+    }
+
+    It 'Passes through non-existent paths unchanged' {
+        $result = @(Resolve-DependencyFiles -Dependencies @("C:\nonexistent\fake.zip") -DestinationPath $destFolder)
+
+        $result | Should -HaveCount 1
+        $result[0] | Should -Be "C:\nonexistent\fake.zip"
+    }
+}

--- a/Tests/DownloadProjectDependencies.Test.ps1
+++ b/Tests/DownloadProjectDependencies.Test.ps1
@@ -630,9 +630,11 @@ Describe "DownloadProjectDependencies - Resolve-DependencyFiles Tests" {
     }
 
     It 'Passes through non-existent paths unchanged' {
-        $result = @(Resolve-DependencyFiles -Dependencies @("C:\nonexistent\fake.zip") -DestinationPath $destFolder)
+        $nonExistentZipFile = Join-Path $testFolder "nonexistent-fake.zip"
+
+        $result = @(Resolve-DependencyFiles -Dependencies @($nonExistentZipFile) -DestinationPath $destFolder)
 
         $result | Should -HaveCount 1
-        $result[0] | Should -Be "C:\nonexistent\fake.zip"
+        $result[0] | Should -Be $nonExistentZipFile
     }
 }

